contrib/generate-ds20.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/python3
+# pylint: disable=invalid-name,missing-docstring
+#
+# Copyright (C) 2022 Richard Hughes <richard@hughsie.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+#
+# pylint: disable=consider-using-f-string
+
+import sys
+import argparse
+import configparser
+import base64
+from typing import List
+
+
+def usage(return_code):
+    """print usage and exit with the supplied return code"""
+    if return_code == 0:
+        out = sys.stdout
+    else:
+        out = sys.stderr
+    out.write(
+        "usage: {} <QUIRK-FILENAME> [--instance-id <INSTANCE-ID>] --bufsz BUFSZ\n".format(
+            sys.argv[0]
+        )
+    )
+    sys.exit(return_code)
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-b",
+        "--bufsz",
+        type=int,
+    )
+    parser.add_argument("-i", "--instance-id", type=str)
```

Comment:
you should add a help string for this

Reply:
Fixed, and I've made it a a bit more resistant to doing the wrong thing.